### PR TITLE
Avoid receiveNoWait when only one subscriber

### PR DESCRIPTION
**Pull request by deguif**
This follows #625 I created yesterday 

Let's try to explain the issue with further details.
This occurs when using AWS SQS as queue system.

The SQS context doesn't support `SubscriptionConsumer` and throws an exception when calling its `createSubscriptionConsumer()` method. This exception is then caught and is replaced by the fallback subscription consumer.

The `FallbackSubscriptionConsumer` is a basic implementation which loops over a list of subscribers and call on each of them their `receiveNoWait()` method.
This is a cost killer for AWS SQS as this will query for new messages without polling.

Here's an example of the command I run and which leads to this dramatic cost:
`php bin/console enqueue:transport:consume --time-limit=+80seconds --receive-timeout=20000 --idle-timeout=5000 MyApp\Processor`
This command is run by a supervising system which rerun it once it stops (reached its time limit).

And here's my proposed solution, which avoids calling `receiveNoWait()` when there is only one subscriber in the fallback subscription consumer.
This is not a full fix as this won't avoid this cost killing when passing multiple subscribers to the fallback subscription consumer.

### CODE REVIEW

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php [hunk 1]
```diff
@@ -13,21 +13,16 @@
...
 
-    public function __construct()
-    {
-        $this->subscribers = [];
-    }
-
     public function consume(int $timeout = 0): void
     {
-        if (empty($this->subscribers)) {
+        if (!$subscriberCount = \count($this->subscribers)) {
```
Comment (makasim) [inline pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: please revert the change, the var could be initialized after this if condition.

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php [hunk 2]
```diff
@@ -41,13 +36,13 @@ public function consume(int $timeout = 0): void
              * @var callable $processor
              */
             foreach ($this->subscribers as $queueName => list($consumer, $callback)) {
-                $message = $consumer->receiveNoWait();
+                $message = 1 === $subscriberCount ? $consumer->receive((int) ($timeout * 1000)) : $consumer->receiveNoWait();
```
Comment (makasim) [inline pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: [timeout should be milliseconds already](https://github.com/queue-interop/queue-interop/blob/master/src/SubscriptionConsumer.php#L9)

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php (hunk 1 above)
Comment (deguif) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: Simple question: isn't `$this->subscribers` always an array?

Comment (makasim) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: always

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php (hunk 2 above)
Comment (deguif) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: It was previously converted to seconds, but I misunderstood the previous operation done on timeout :)

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php [hunk 3]
```diff
@@ -13,21 +13,16 @@
...
 
     /**
      * @var int
      */
     private $idleTime = 0;
 
-    public function __construct()
-    {
-        $this->subscribers = [];
-    }
```
[the remark this answers is not in the document]
Comment (deguif) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: Ok, but just curious what it's done this way? CS?

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php (hunk 1 above)
Comment (deguif) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: So are you sure you want to keep the `empty` check, as `count` should do the same, and the advantage is that it's allowed to initialize the variable at the same time ;)

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php (hunk 2 above)
Comment (makasim) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: which operation are you referring to?

Comment (deguif) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: https://github.com/php-enqueue/enqueue-dev/pull/626/files/f524a099908afa284278ef337b4042339094bf6b#diff-5f17e0abec53c03fec471b2c5f9f668dL34

**Comment by makasim** [comment]
could you please rename timeout argument to `timeoutMs` and use it in `receive` method.

**Comment by deguif** [comment]
Done

File: pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (deguif) [reply pkg/enqueue/Consumption/FallbackSubscriptionConsumer.php]: Done

**Comment by makasim** [comment]
Thank you @deguif !

**Comment by deguif** [comment]
You're welcome.
